ghastly/ghastly.py

```diff
@@ -0,0 +1,315 @@
+import numpy as np
+import openmc
+import ghastly
+from jinja2 import Environment, PackageLoader
+from ghastly import read_input
+from ghastly import region
+from lammps import lammps
+
+env = Environment(loader=PackageLoader('ghastly', 'templates'))
+
+
+def fill_core(input_file, rough_pf):
+    '''
+    This function uses OpenMC's model.pack_spheres function to "rough" pack
+    the full core region defined by the Ghastly input file, then uses a series
+    of Jinja templates to create a LAMMPS input file that will run a physics
+    simulation, modeling not only the settling of the roughly packed pebbles
+    due to gravtiy, but also handle the addtion of any extra pebbles needed to
+    reach the target number of pebbles needed.
+
+    Parameters
+    ----------
+    input_file : str
+        Name of ghastly input file, which should be in JSON format, following
+        the layout and parameter naming convention given in the example inputs.
+    rough_pf : float
+        Packing fraction used in OpenMC packing function.  Choosing a rough_pf
+        greater than the target pf in the input file can result in overfilling
+        of the core, and so it is reccommended to use a rough_pf that is
+        less than or equal to the desired final pf.
+
+    Returns
+    -------
+    While this function does not have a return at its completion, it will
+    generate a number of files - both LAMMPS input files and files LAMMPS
+    itself will generate.  For that reason, it's recommended that this be
+    run inside its own directory, in order to keep these files organized.
+    '''
+
+    input_block = read_input.InputBlock(input_file)
+    sim_block = input_block.create_obj()
+
+    rough_pack = []
+    pack_zones = sim_block.core_main | sim_block.core_outtake
+    core_volume = sum([i.volume for i in pack_zones.values()])
+    for element in pack_zones.values():
+        if type(element) == ghastly.core.CylCore:
+            coords = pack_cyl(sim_block, element, rough_pf)
+            rough_pack += coords
+        else:
+            pass
+
+    n_pebbles = int((sim_block.pf*core_volume)/sim_block.pebble_volume)
+    print("Equivalent number of pebbles is "+str(n_pebbles))
+
+    pebbles_left = n_pebbles - len(rough_pack)
+
+    if pebbles_left < 0:
+        raise ValueError('''Negative pebbles left - core overfilled.  Reduce
+                         rough_pf and try again.''')
+
+    x_b, y_b, z_b = find_box_bounds(sim_block, pour=True)
+
+    fake_dump_file(rough_pack, "rough-pack.txt", "ff ff ff",
```

Comment:
This should probably be called coords_to_lammps_dumpfile or similiar. It would be ok to have it be "write_dumpfile" though it may be confusing as that implies it would be a ghastly dumpfile. Maybe "write_lammps_dumpfile" is sufficient.